### ISSUE TITLE
Hide missing files in Library View

### ISSUE THREAD

**Issue by captn3m0** [johnfactotum/foliate#582]
**Is your feature request related to a problem? Please describe.**

Too many entries in the library are for files that are no longer on my system. I don't want to see entries for books that I abandoned/removed from my system.

**Describe the solution you'd like**
A checkbox in preferences that says 

- [ ] Hide missing books from Library

Once this is checked, only files that are still present on disk and accessible are shown in the library.

**Describe alternatives you've considered**

- A button that "cleans" the library and removes missing books
- An option to only limit my library to specific directories. If I open a file from an arbitrary directory that is not in a known Library location, it should not be added to my library. For eg, I open books from the temporary mozilla download directory (`~/tmp/mozilla_tmp`) - adding them to the library is meaningless to me, since I don't expect to re-open them.

**Comment by johnfactotum** [comment]
Yes, I agree. A way to clean up the library is definitely needed.

> For eg, I open books from the temporary mozilla download directory (~/tmp/mozilla_tmp) - adding them to the library is meaningless to me, since I don't expect to re-open them.

This is basically https://github.com/johnfactotum/foliate/issues/471.
